web-common/src/features/dashboards/stores/get-explore-state-from-yaml-config.ts

```diff
@@ -31,6 +37,39 @@ export function getExploreStateFromYAMLConfig(
   };
 }
 
+export function createYAMLConfigExploreUrlSearch(
+  validSpecQuery: ReturnType<typeof useExploreValidSpec>,
+  fullTimeRangeQuery: CreateQueryResult<V1MetricsViewTimeRangeResponse>,
+) {
+  return derived(
+    [validSpecQuery, fullTimeRangeQuery],
+    ([validSpec, fullTimeRange]) => {
+      const metricsViewSpec = validSpec.data?.metricsView ?? {};
+      const exploreSpec = validSpec.data?.explore ?? {};
+      const timeRangeSummary = fullTimeRange.data?.timeRangeSummary;
+
+      const exploreStateFromYAMLConfig = getExploreStateFromYAMLConfig(
+        exploreSpec,
+        timeRangeSummary,
+      );
+
+      const timeControlState = getTimeControlState(
+        metricsViewSpec,
+        exploreSpec,
+        timeRangeSummary,
+        exploreStateFromYAMLConfig as ExploreState,
+      );
+
+      const urlParams = convertPartialExploreStateToUrlParams(
+        exploreSpec,
+        exploreStateFromYAMLConfig,
+        timeControlState,
+      );
+      return `?${urlParams.toString()}`;
+    },
+  );
+}
```

Comment:
We have the `exploreStateFromYAMLConfig` in the `DashboardStateDataLoader`. Can we use that, rather than create the same object again?

Reply:
Bookmark is outside of dashboard component tree. So we would have to create `DashboardStateDataLoader` and pass it down. This would mean we would have to either refactor `DashboardStateManager` to optionally get it from parent or always get from parent. Both feels unnecessary, since `getExploreStateFromYAMLConfig` is pretty lightweight.